modules/product/src/test/java/com/opengamma/strata/product/index/OvernightFutureOptionPositionTest.java

```diff
@@ -0,0 +1,122 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.Offset.offset;
+
+import org.junit.jupiter.api.Test;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.product.PortfolioItemSummary;
+import com.opengamma.strata.product.PortfolioItemType;
+import com.opengamma.strata.product.PositionInfo;
+import com.opengamma.strata.product.ProductType;
+
+/**
+ * Test {@link OvernightFutureOptionPosition}.
+ */
+public class OvernightFutureOptionPositionTest {
+
+  private static final ReferenceData REF_DATA = ReferenceData.standard();
+  private static final PositionInfo POSITION_INFO = PositionInfo.builder()
+      .id(StandardId.of("A", "B"))
+      .build();
+  private static final PositionInfo POSITION_INFO2 = PositionInfo.builder()
+      .id(StandardId.of("A", "C"))
+      .build();
+  private static final double QUANTITY = 10;
+  private static final OvernightFutureOption PRODUCT = OvernightFutureOptionTest.sut();
+  private static final OvernightFutureOption PRODUCT2 = OvernightFutureOptionTest.sut2();
+
+  //-------------------------------------------------------------------------
+  @Test
+  public void test_builder_resolved() {
+    OvernightFutureOptionPosition test = sut();
+    assertThat(test.getProduct()).isEqualTo(PRODUCT);
+    assertThat(test.getInfo()).isEqualTo(POSITION_INFO);
+    assertThat(test.getLongQuantity()).isCloseTo(QUANTITY, offset(0d));
+    assertThat(test.getShortQuantity()).isCloseTo(0d, offset(0d));
+    assertThat(test.getQuantity()).isCloseTo(QUANTITY, offset(0d));
+    assertThat(test.withInfo(POSITION_INFO).getInfo()).isEqualTo(POSITION_INFO);
+    assertThat(test.withQuantity(129).getQuantity()).isCloseTo(129d, offset(0d));
+  }
+
+  //-------------------------------------------------------------------------
+  @Test
+  public void test_summarize() {
+    OvernightFutureOptionPosition test = sut();
+    PortfolioItemSummary expected = PortfolioItemSummary.builder()
+        .id(POSITION_INFO.getId().orElse(null))
+        .portfolioItemType(PortfolioItemType.POSITION)
+        .productType(ProductType.OVERNIGHT_FUTURE_OPTION)
+        .currencies(Currency.USD)
+        .description("OvernightFutureOption x 10")
+        .build();
+    assertThat(test.summarize()).isEqualTo(expected);
+  }
+
+  //-------------------------------------------------------------------------
+  @Test
+  public void test_withQuantity() {
+    OvernightFutureOptionPosition base = sut();
+    double quantity = 75343d;
+    OvernightFutureOptionPosition computed = base.withQuantity(quantity);
+    OvernightFutureOptionPosition expected = OvernightFutureOptionPosition.builder()
+        .info(POSITION_INFO)
+        .product(PRODUCT)
+        .longQuantity(quantity)
+        .build();
+    assertThat(computed).isEqualTo(expected);
```

Comment:
maybe lets check also the logic for shortQuantity?

Reply:
Added